scripts/monitor/visualization/resource_monitoring.py

```diff
@@ -0,0 +1,596 @@
+from typing import Dict
+import matplotlib.pyplot as plt
+from bokeh.models import Title
+from matplotlib.backends.backend_pdf import PdfPages
+import numpy
+import pandas as pd
+import pandas_bokeh
+from operator import truediv
+
+
+class ResourceUsage:
+    instances = 0  # count number of class instances (can't find a way to update this when an instance is deleted)
+
+    def __init__(self, meta_runtime: pd.DataFrame, metrics: pd.DataFrame) -> None:
+        """
+        initialization step: bring meta, runtime, and metrics data into scope
+        :param meta_runtime: meta_runtime table
+        :param metrics: metrics table
+        """
+        self.meta_runtime = meta_runtime
+        self.metrics = metrics
+        self.instances += 1
+
+    def run(self) -> None:  # maybe do this as part of initialization
+        """
+        Do everything that needs to be done to set up the webpage, and then set up the webpage
```

Comment:
? set up then set up?

Reply:
This was poor wording, I've fixed it now- I meant that it will first load all the files the webpage uses, and then set up the webpage